src/Tribe/Collisions/Average_Proximity_Start_Detector.php

```diff
@@ -0,0 +1,164 @@
+	 * @var int
+	 */
+	protected $margin;
+
+	/**
+	 * Tribe__Collisions__Average_Proximity_Start_Detector constructor.
+	 *
+	 * @param int $margin
+	 */
+	public function __construct( $margin = 0 ) {
+		$this->margin = $margin;
+	}
+
+	/**
+	 * Computes the collision-based difference of two or more arrays of segments returning an array of elements from
+	 * the first array not colliding with any element from the second array according to the collision detection
+	 * strategy implemented by the class.
+	 *
+	 * Note: points are segments with matching start and end.
+	 *
+	 * @param array $a     An array of elements each defining the start and end of a segment in the format [<start>,
+	 *                     <end>].
+	 * @param array $b,... An array (ore more arrays) of elements each defining the start and end of a segment in the
+	 *                     format [<start>, <end>].
+	 *
+	 * @return array An array of elements each defining the start and end of a segment in the format [<start>, <end>].
+	 */
+	public function diff( array $a, array $b ) {
+		$bs = func_get_args();
+		$a = array_shift( $bs );
+
+		$bs = array_filter( $bs );
+		if ( empty( $bs ) ) {
+			return $a;
+		}
+
+		$intersected = call_user_func_array( array( $this, 'intersect' ), func_get_args() );
+
+		$diffed = array();
+
+		foreach ( $a as $candidate ) {
+			if ( ! in_array( $candidate, $intersected ) ) {
+				$diffed[] = $candidate;
+			}
+		}
+
+		return $diffed;
+	}
+
+	/**
+	 * Computes the collision-based intersection of two or more arrays of segments returning an array of elements from
+	 * the first array colliding with at least one element from the second array according to the collision detection
+	 * strategy implemented by the class.
+	 * If more than one array is specified then a segment from the first array has to collide with at least one segment
+	 * from each intersecting segment.
+	 * If a lax intersection is needed use `touch`.
+	 *
+	 * Note: points are segments with matching start and end.
+	 *
+	 * @see Tribe__Collisions__Detection_Strategy::touch()
+	 *
+	 * @param array $a     An array of elements each defining the start and end of a segment in the format [<start>,
+	 *                     <end>].
+	 * @param array $b,... An array (ore more arrays) of elements each defining the start and end of a segment in the
+	 *                     format [<start>, <end>].
+	 *
+	 * @return array An array of arrays of elements each defining the start and end of a segment in the format
+	 *               [<start>, <end>]; the first array contains the segments of $a that collided while the second array
+	 *               contains the segments that did collide with each colliding element of $a
+	 */
+	public function report_intersect( array $a, array $b ) {
+		$reported = call_user_func_array( array( 'parent', 'report_intersect' ), func_get_args() );
+
+		if ( empty( $reported ) ) {
+			return array();
+		}
+
+		$as = reset( $reported );
```

Comment:
`a_segments` and `b_segments`?
PHP `array_diff` uses `array1` and `array2` but that's not so much better... `segment_1` and `segment_2`?
My brain is jelly right now